packages/recoil/recoil_values/Recoil_atom.js

```diff
@@ -292,6 +293,23 @@ function baseAtom<T>(options: BaseAtomOptions<T>): RecoilState<T> {
         return getLoadable(recoilValue).toPromise();
       }
 
+      function observe<S>(
+        recoilValue: RecoilValue<S>,
```

Comment:
Needs to handle the case when `recoilValue` is actually ourselves.  Either this could redirect to `onSet()` in that case or `onSet()` could be renamed/generalized to handle any node and then a `onSet()` is just a wrapper calling that.

Reply:
Ahh, right - I updated this to check for key equality and throw an error if if's a self-observe.